packages/twenty-server/src/core/quick-actions/quick-actions.service.ts

```diff
@@ -0,0 +1,147 @@
+import { Injectable } from '@nestjs/common';
+
+import { v4 as uuidv4 } from 'uuid';
+
+import { Record as IRecord } from 'src/workspace/workspace-query-builder/interfaces/record.interface';
+
+import { isWorkEmail } from 'src/utils/is-work-email';
+import { stringifyWithoutKeyQuote } from 'src/workspace/workspace-query-builder/utils/stringify-without-key-quote.util';
+import { WorkspaceQueryRunnerService } from 'src/workspace/workspace-query-runner/workspace-query-runner.service';
+import { IntelligenceService } from 'src/core/quick-actions/intelligence.service';
+
+@Injectable()
+export class QuickActionsService {
+  constructor(
+    private readonly workspaceQueryRunnunerService: WorkspaceQueryRunnerService,
+    private readonly intelligenceService: IntelligenceService,
+  ) {}
+
+  async createCompanyFromPerson(id: string, workspaceId: string) {
+    const person = (
+      (await this.workspaceQueryRunnunerService.executeAndParse(
+        `query {
+        personCollection(filter: {id: {eq: "${id}"}}) {
+              edges {
+                node {
+                  id
+                  email
+                  companyId
+                }
+              }
+            }
+          }
+        `,
+        'person',
+        '',
+        workspaceId,
+      )) as IRecord
+    ).edges?.[0]?.node;
+
+    if (!person.companyId && person.email && isWorkEmail(person.email)) {
+      const companyDomainName = person.email.split('@')[1];
+      const companyNameSmallCase = companyDomainName.split('.')[0];
+      let relatedCompanyId = uuidv4();
+
+      const existingCompany =
+        (await this.workspaceQueryRunnunerService.executeAndParse(
+          `query {companyCollection(filter: {domainName: {eq: "${companyDomainName}"}}) {
+                edges {
+                  node {
+                    id
+                  }
+                }
+              }
+            }
+          `,
+          'company',
+          '',
+          workspaceId,
+        )) as IRecord;
+
+      if (existingCompany.edges?.length) {
+        relatedCompanyId = existingCompany.edges[0].node.id;
+      }
+
+      await this.workspaceQueryRunnunerService.execute(
+        `mutation {
+          insertIntocompanyCollection(objects: ${stringifyWithoutKeyQuote([
+            {
+              id: relatedCompanyId,
+              name:
+                companyNameSmallCase.charAt(0).toUpperCase() +
```

Comment:
@FelixMalfait I think we already have an util for capitalization

Reply:
Just fixed it at the same time you commented! Thanks!